core/client/fs/src/test/java/alluxio/client/block/stream/BlockWorkerDataReaderTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.stream;
+
+import static org.mockito.ArgumentMatchers.any;
+
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.options.InStreamOptions;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.util.ConfigurationUtils;
+import alluxio.util.FileSystemOptions;
+import alluxio.util.io.BufferUtils;
+import alluxio.wire.BlockReadRequest;
+import alluxio.wire.FileInfo;
+import alluxio.worker.block.BlockWorker;
+import alluxio.worker.block.io.MockBlockReader;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Collections;
+import java.util.Random;
+
+/**
+ * Unit tests for {@link BlockWorkerDataReader}.
+ */
+public class BlockWorkerDataReaderTest {
+  private static final long BLOCK_ID = 1L;
+  private static final int CHUNK_SIZE = 128;
+  private static final Random RANDOM = new Random();
+
+  private BlockWorker mBlockWorker;
+  private BlockWorkerDataReader.Factory mDataReaderFactory;
+
+  @Before
+  public void before() throws Exception {
+    mBlockWorker = Mockito.mock(BlockWorker.class);
```

Comment:
Consider using `NoopBlockWorker`?  Perhaps we should rename it to `MockBlockWorker`